hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestClientScannerTimeouts.java

```diff
@@ -101,6 +102,9 @@ public static void setUpBeforeClass() throws Exception {
     conf.setInt(HConstants.HBASE_CLIENT_SCANNER_TIMEOUT_PERIOD, scanTimeout);
     conf.setInt(HBASE_CLIENT_META_READ_RPC_TIMEOUT_KEY, metaScanTimeout);
     conf.setInt(HBASE_CLIENT_META_SCANNER_TIMEOUT, metaScanTimeout);
+    // set to true by default here. it only affects next() calls, and we'll explicitly
+    // set it to false in one of the tests below to test legacy behavior for next call
+    conf.setBoolean(HBASE_CLIENT_USE_SCANNER_TIMEOUT_FOR_NEXT_CALLS, true);
```

Comment:
This is perhaps too subtle -- we're no longer testing with the default configuration except for a single test. I'd have more confidence in the coverage if, for example, this class was parameterized on this config flag.